pombola/south_africa/views/download.py

```diff
@@ -0,0 +1,86 @@
+import os
+
+from django.db.models import Prefetch, Q
+from django.http import Http404, StreamingHttpResponse
+from django.shortcuts import get_object_or_404
+from django.views.generic import TemplateView
+
+import xlsx_streaming
+from pombola.core.models import Contact, Organisation, Person, Position
+
+MP_DOWNLOAD_TEMPLATE_SHEET = os.path.join(
+    os.path.dirname(os.path.realpath(__file__)), "mp-download-template.xlsx"
+)
+
+
+def get_email_address_for_person(person):
+    if person.email:
+        return person.email
+    if len(person.email_addresses) > 0:
+        return person.email_addresses[0].value
+    return ""
+
+
+def download_members_xlsx(request, slug):
+    organisation = get_object_or_404(Organisation, slug=slug)
+
+    # Get all of the currently_active positions at the organisation
+    organisation_positions = (
+        Position.objects.currently_active()
+        .filter(organisation=organisation)
+        .values("id")
+    )
+
+    party_positions = (
+        Position.objects.currently_active()
+        .filter(title__slug="member")
+        .filter(organisation__kind__slug="party")
+        .select_related("organisation")
+    )
+
+    cell_phone_contacts = Contact.contact_number_contacts()
+    email_contacts = Contact.email_contacts()
+
+    # Get the persons from the positions
+    persons = (
+        Person.objects.filter(hidden=False)
+        .filter(position__id__in=organisation_positions)
+        .distinct()
+        .prefetch_related(
+            "alternative_names",
+            Prefetch(
+                "position_set",
+                queryset=party_positions,
+                to_attr="active_party_positions",
+            ),
+            Prefetch("contacts", queryset=cell_phone_contacts, to_attr="cell_numbers"),
+            Prefetch("contacts", queryset=email_contacts, to_attr="email_addresses"),
+        )
+    )
+
+    def person_row_generator():
+        for person in persons:
+            email = get_email_address_for_person(person)
+            yield (
+                person.name,
+                person.cell_numbers[0].value if len(person.cell_numbers) > 0 else "",
```

Comment:
how about including all numbers separated by comma and space?

I suggested just space for email addresses so people don't accidentally copy and paste the comma and fail but for numbers it should be fine